lib/mocha.js

```diff
@@ -37,7 +37,7 @@ exports = module.exports = Mocha;
  * These are the states it can be in.
  * @private
  */
-var mochaStates = utils.defineConstants({
+var mochaStates = Object.freeze({
```

Comment:
Can we use `const` instead of `var`, does it get transpiled?

Reply:
We can, it does get transpiled. But I think I'd prefer to do everything at once with lebab instead of trickling those in in unrelated changes

Reply:
I'm unsure how I feel about this... 

On the one hand, we've done huge sweeping changes (e.g., prettier, eslint) before. But this makes `git blame` less useful.

On the other, this will look inconsistent and weird if we change things piecemeal.

I'd like to get the opinion of @mochajs/core on how we should proceed with "modernizing" our codebase

Reply:
I guess modernizing our code base won't be a big bang. We should include smaller ES6 changes in our daily PR's. There should be dedicated PR's for bigger changes as e.g. turning runner's callback hell into something more pleasant.
So above not changing `var` to `const` looks odd and IMO should be changed.